### ISSUE TITLE
KeyError when parsing streamline data

### ISSUE THREAD

**Issue by m-col** [BrancoLab/BrainRender#19]
Lines [182](https://github.com/BrancoLab/BrainRender/blob/master/BrainRender/Utils/parsers/streamlines.py#L182) and Lines [192](https://github.com/BrancoLab/BrainRender/blob/master/BrainRender/Utils/parsers/streamlines.py#L192) of BrainRender/Utils/parsers/streamlines.py use the int `0` to get data: `data['injection_sites'][0]`

I had to change this to the string `'0'`: `data['injection_sites']['0']` otherwise I would get a key error.

I assume using the int has worked in the past, so perhaps a code change has altered how the data is being read from its JSON file.

Does the int work for you currently?

**Comment by FedeClaudi** [comment]
mmm, never had this problem before. 

If I run `StreamlinesScene` from `Examples\example_scenes.py` it works ok. 
In this function I get the data through the `streamlines_api`:
```streamlines_files, data = streamlines_api.download_streamlines_for_region("PAG") ```
which fetches the data from the cache at https://neuroinformatics.nl via a request, the result of which is turned into `json`.

But passing downloaded data works too:
```
from BrainRender.Utils.data_io import listdir
from BrainRender.scene import Scene

scene = Scene()
streamlines_files = listdir("Examples/example_files/streamlines")
scene.add_streamlines(sl_file=streamlines_files, color="green")
scene.render()

```

Can you send a screenshot of the code causing problems?

**Comment by m-col** [comment]
This minimal piece code triggers the error on the last line:

```
from BrainRender.scene import Scene
from BrainRender.Utils import parsers

scene = Scene()
filepaths, _ = parsers.streamlines.StreamlinesAPI().download_streamlines_for_region('VM')
scene.add_streamlines(
    filepaths,
    color="green",
)
```

And the output, trace and error message:

```
  0%|                                                                                                         | 0/5 [00:00<?, ?it/s]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/mcol/git/BrainRender/BrainRender/scene.py", line 671, in add_streamlines
    streamlines = parse_streamline(filepath=slf, *args, **kwargs)
  File "/home/mcol/git/BrainRender/BrainRender/Utils/parsers/streamlines.py", line 182, in parse_streamline
    lines_data = data['lines'][0]
KeyError: 0
```

If the `0` key is on line 182 is replaced with `'0'`, then line 192 triggers it. It then works if they are both `'0'`.

Does the above code for you?

**Comment by FedeClaudi** [comment]
Oh I see now.

When downloading the streamlines for a region you are return `filepaths` and `data`. Currently you're discarding data and passing `filepaths` to the scene. That causes an error. 
This edited code works:

```
scene = Scene()
filepaths, data = parsers.streamlines.StreamlinesAPI().download_streamlines_for_region('VM')
scene.add_streamlines(
    data,
    color="green",
)
```

It should work with both though, I will fix it.

**Comment by FedeClaudi** [comment]
Fixed with 371f9e26673708ce7928d7dd2c1c82ac9b57dfe4

**Comment by m-col** [comment]
I see - OK cool, thanks. I guess I should have used the data directly.
